gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -378,19 +378,37 @@ gst_tensor_converter_set_property (GObject * object, guint prop_id,
     const GValue * value, GParamSpec * pspec)
 {
   GstTensorConverter *self;
+  GstTensorsInfo *info;
+  guint num;
+  const gchar *value_str;
 
   self = GST_TENSOR_CONVERTER (object);
+  info = &self->tensors_info;
 
   switch (prop_id) {
     case PROP_INPUT_DIMENSION:
-      if (gst_tensor_parse_dimension (g_value_get_string (value),
-              self->tensor_info.dimension) == 0)
-        GST_WARNING ("input dimension unknown (optional).");
+      value_str = g_value_get_string (value);
+      num = gst_tensors_info_parse_dimensions_string (info, value_str);
+
+      if (num == 0) {
+        GST_WARNING ("%s is invalid dimension string.", value_str);
+      } else if (info->num_tensors > 0 && info->num_tensors != num) {
+        GST_WARNING ("%s, the number of tensor is %u.", value_str, num);
+      }
+
+      info->num_tensors = num;
```

Comment:
Q. Is it okay to perform `info->num_tensors = num` even when `num == 0`? IMO, it's somewhat weird because `info` will be not changed in `gst_tensors_info_parse_dimensions_string()` in this case.

Reply:
info (self->tensors_info) will be updated if value string is not null.
Actually if developer sets invalid value string (type or dim) in pipeline, tensor-converter may raise an error or work improperly.

Reply:
Well.. please refer to the below codes to explain why I thought like that.

https://github.com/nnstreamer/nnstreamer/blob/f7b16f3127235f59479b04e3c10ada7a1ca075d4/gst/nnstreamer/tensor_common.c#L380-L382

Even if value_str is not null, `info` is not changed when its dims (i.e., `num_dims`) are evaluated to 0.
Please let me know if I'm wrong. And, I also totally agree with your thought about an invalid string.

Reply:
You mean..about the initialization when user changes the property?
(old number 3 > update 2 dimensions, then count is 2 but the array with index 2 has not empty)

Reply:
Resolved in offline discussion :)

Reply:
I will add initialization for dimensions and types.
```
      for (i = num; i < NNS_TENSOR_SIZE_LIMIT; ++i) {
        info->info[i].type = _NNS_END;
      }
```